docs/beyond_basics/101-182-catalog.rst

```diff
@@ -3,7 +3,573 @@
+""""""""""""""""""""""""""
+
+Let's dive into it and create our own catalog! We'll start by creating and activating a new and empty virtual environment:
+
+.. code-block:: bash
+
+   $ python -m venv my_catalog_env
+   $ source my_catalog_env/bin/activate
+
+Then we can install ``datalad-catalog`` with ``pip``. This process will also install ``datalad`` and other dependencies:
+
+.. code-block:: bash
+
+   $ pip install datalad-catalog
+
+After that, you can check the installation by running the ``datalad catalog`` command with the ``--help`` flag:
+
+.. runrecord:: _examples/DL-101-182-101
+   :language: console
+   :workdir: catalog
+   :cast: catalog_basics
+   :notes: Let's test the installation and look at the help information
+   
+   $ datalad catalog --help
+
+At this stage, you might be wondering why the catalog command is preceded by ``datalad`` as in ``datalad catalog``.
+DataLad Catalog is an extension of DataLad, which provides base functionality that the catalog generation process uses.
+It is installed as a dependency during the installation of DataLad Catalog, and provides supporting functionality during
+the catalog generation process.
+
+The main catalog functionality
+""""""""""""""""""""""""""""""
+As you likely saw in the ``--help`` information, DataLad Catalog has several main commands to support
+the process of catalog generation. These include:
+
+- ``create``: create a new catalog
+- ``add``: add metadata to a catalog
+- ``remove``: remove metadata from a catalog
+- ``serve``: serve the catalog locally on an http server for testing purposes
+- ``set-super``: set the so-called super-dataset of the catalog, i.e. the dataset that will be displayed as the catalog's home page
+- ``validate``: validate metadata according to the catalog schema
+- ``workflow-new``: run a multi-step workflow for extracting and converting metadata from a super-dataset and its subdatasets, and adding these to a newly created catalog
+- ``workflow-update``: run a multi-step workflow for extracting and converting metadata from a new subdataset of an existing super-dataset, and adding these to an existing catalog
+
+Creating a new catalog
+""""""""""""""""""""""
+
+With the ``create`` subcommand, you can create a new catalog. Let's try it out!
+
+.. runrecord:: _examples/DL-101-182-102
+   :language: console
+   :workdir: catalog
+   :cast: catalog_basics
+   :notes: Let's test the installation and look at the help information
+   
+   $ datalad catalog create --catalog-dir data-cat
+
+The catalog ``create(ok)`` result shows that the catalog was successfully created at the specified location (``./data-cat``),
+wich was passed to the command with the ``-c/--catalog-dir`` flag.
+
+Now we can inspect the catalog's content with the ``tree`` command:
+
+.. runrecord:: _examples/DL-101-182-103
+   :language: console
+   :workdir: catalog
+   :cast: catalog_basics
+   :notes: We can inspect the catalog's content with the tree command
+   
+   $ tree data-cat
```

Comment:
I think the output could be trimmed here too (the main aspect is the set of subdirectories, not the files). In addition, consider adding a `:linereplace:` expression to remove the version numbers that will change all the time.